controllers/drplacementcontrol_controller.go

```diff
@@ -964,45 +987,49 @@ func (d *DRPCInstance) runInitialDeployment() (bool, error) {
 	d.advanceToNextDRState()
 
 	d.log.Info(fmt.Sprintf("DRPC (%+v)", d.instance))
-	setMetricsTimerFromDRState(rmn.Deployed, timerStop)
+	d.setMetricsTimerFromDRState(rmn.Deployed)
 
 	return done, nil
 }
 
-func setMetricsTimerFromDRState(stateDR rmn.DRState, stateTimer timerState) {
+func (d *DRPCInstance) setMetricsTimerFromDRState(stateDR rmn.DRState) {
 	switch stateDR {
 	case rmn.FailingOver:
-		setMetricsTimer(&failoverTime, stateTimer, stateDR)
+		d.setMetricsTimer(&failoverTime, timerStart, stateDR)
+	case rmn.FailedOver:
+		d.setMetricsTimer(&failoverTime, timerStop, stateDR)
 	case rmn.FailingBack:
-		setMetricsTimer(&failbackTime, stateTimer, stateDR)
+		d.setMetricsTimer(&failbackTime, timerStart, stateDR)
+	case rmn.FailedBack:
+		d.setMetricsTimer(&failbackTime, timerStop, stateDR)
 	case rmn.Relocating:
-		setMetricsTimer(&relocateTime, stateTimer, stateDR)
+		d.setMetricsTimer(&relocateTime, timerStart, stateDR)
+	case rmn.Relocated:
+		d.setMetricsTimer(&relocateTime, timerStop, stateDR)
 	case rmn.Deploying:
-		setMetricsTimer(&deployTime, stateTimer, stateDR)
-	case rmn.FailedBack:
-		fallthrough
-	case rmn.FailedOver:
-		fallthrough
+		d.setMetricsTimer(&deployTime, timerStart, stateDR)
 	case rmn.Deployed:
-		fallthrough
-	case rmn.Relocated:
-		fallthrough
+		d.setMetricsTimer(&deployTime, timerStop, stateDR)
 	default:
 		// not supported
 	}
 }
 
-func setMetricsTimer(wrapper *timerWrapper, desiredTimerState timerState, reconcileState rmn.DRState) {
+func (d *DRPCInstance) setMetricsTimer(
+	wrapper *timerWrapper, desiredTimerState timerState, reconcileState rmn.DRState) {
 	switch desiredTimerState {
 	case timerStart:
-		if reconcileState != wrapper.reconcileState {
-			wrapper.reconcileState = reconcileState
-			wrapper.timer = *prometheus.NewTimer(prometheus.ObserverFunc(wrapper.gauge.Set))
+		if reconcileState != d.metricsTimer.reconcileState {
+			d.metricsTimer.timer.ObserveDuration() // stop gauge timer in case one is still running
+
+			d.metricsTimer.reconcileState = reconcileState
+			d.metricsTimer.timer = *prometheus.NewTimer(
+				prometheus.ObserverFunc(wrapper.gauge.WithLabelValues(d.instance.Name).Set))
 		}
 	case timerStop:
-		wrapper.timer.ObserveDuration()
-		wrapper.histogram.Observe(wrapper.timer.ObserveDuration().Seconds()) // add timer to histogram
-		wrapper.reconcileState = rmn.Deployed                                // "reserved" value
+		d.metricsTimer.timer.ObserveDuration()                                      // stop gauge timer
+		wrapper.histogram.Observe(d.metricsTimer.timer.ObserveDuration().Seconds()) // add timer to histogram
+		d.metricsTimer.reconcileState = rmn.Deployed                                // "reserved" value
```

Comment:
Why do you have to set the reconcile state to `Deployed`? Why don't you assign that actual state that was called for, similar to line 1025?

Reply:
After refactoring this function to just look at the reconciler state and then derive the timer state from that, this is no longer needed